polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -187,29 +187,33 @@ pub mod pallet {
 	pub type BalanceOf<T> =
 		<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
 
-	#[pallet::config]
+	#[pallet::config(with_default)]
 	/// The module configuration trait.
 	pub trait Config: frame_system::Config {
-		/// The overarching event type.
-		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
-
-		/// A lockable currency.
 		// TODO: We should really use a trait which can handle multiple currencies.
+		#[pallet::no_default]
+		/// A lockable currency.
 		type Currency: LockableCurrency<Self::AccountId, Moment = BlockNumberFor<Self>>;
 
 		/// The `MultiAsset` matcher for `Currency`.
+		#[pallet::no_default_bounds]
 		type CurrencyMatcher: MatchesFungible<BalanceOf<Self>>;
 
+		// TODO: Would love to add a default for this, but I need access to `RuntimeOrigin`
+		// from the derive_impl
+		#[pallet::no_default]
```

Comment:
What is usually the default that you would want to provide?

Reply:
You should be able to use `no_default_bounds` for such cases.